Add comms on migration to single-region buckets (#58)
malariagen/vobs-updates: _posts/2024-11-19-single-region-transition.md

---
layout: post

> title:  "Update on Data Storage"

Reviewer: ```suggestion
title:  "Migration to single-region GCS buckets"
```

tags: data
---

We are making schanges to how data is stored in the Vector Observatory (VObs).

> Currently, data is stored in multi-region Google Cloud Storage (GCS) buckets. After evaluating accessibility needs, we’ve decided to move the data to single-region buckets located in the United States (US). This transition reduces our running costs without affecting data availability or performance.

Reviewer: Maybe be more specific about the region. Also not obvious what "after evaluating accessibility needs" means, maybe can leave out?

> Due to this, we are starting to roll-out some changes:

Reviewer: ```suggestion
To access data in the new single-region buckets, please follow the instructions below:
```

> 1. **If you access data via the malariagen_data API:**

Reviewer: ```suggestion
1. **If you access data via the malariagen_data Python API:**
```

>   - Please ensure that you are only using API versions v13.0.0 or newer. If you install `malariagen_data` without specifying a version (e.g., using `%pip install malariagen_data`), the latest version will automatically be installed.

Reviewer: ```suggestion
  - Please ensure that you are only using API versions v13.0.0 or newer. If you install `malariagen_data` without specifying a version (e.g., using `pip install malariagen_data`), the latest version will automatically be installed. If you have previously installed `malariagen_data`, please run `pip install --upgrade malariagen_data` to upgrade to the latest software version.
```

  - To check the version you’re currently using in a notebook, run:
  
{% highlight python %}
import malariagen_data

malariagen_data.__version__

{% endhighlight %}

>   - **If you use Google Colab:**
>       - Confirm you’re running the latest version of the API (v13.0.0 or newer).
>       - If you encounter an error due to being allocated a Virtual Machine (VM) outside of the US, go to `"Runtime" > "Disconnect and delete runtime"` to allocate a new VM within the US region.

Reviewer: ```suggestion
```

Is this strictly necessary here?

2. **If you access data via data directly from GCS buckets:**
- The following bucket name changes have been implemented:
    - Ag data previously stored in `gs://vo_agam_release` is now in `gs://vo_agam_release_master_us_central1`
    - Af data previously stored in `gs://vo_afun_release` is now in  `gs://vo_afun_release_master_us_central1`

> - All data has been mirrored to the new buckets. There is no need to download any data.

Reviewer: ```suggestion
- All data has been mirrored to the new buckets. Please update any scripts or code to use the new bucket names.
```

> - The multi-region buckets (`vo_agam_release`, `vo_agam_release`) will be deprecated by November 30th, 2024. You might need to update any scripts accessing these buckets to use the new names.
> - All new data will only be available in the single-region buckets moving forward.

Reviewer: ```suggestion
**N.B., the multi-region buckets (`vo_agam_release`, `vo_agam_release`) will be deprecated on November 30th, 2024. Please follow the instructions above to migrate to using single-region buckets before this date.**
```

Maybe make this a bit bolder so it's harder to miss?
